### PR TITLE
Remove the std type_trait polyfills

### PR DESCRIPTION
**Pull request by MarkCMann**
We no longer need these polyfills so let's remove them

### CODE REVIEW

**Review by janondrusek — APPROVED** [review]
Glanced over all the changed files, seems to be a consistently applied change.

**Review by lehecka — COMMENTED** [review]
is it safe to remove the usage of absl::disjunction?
The FB4A had been always the blocker for std::optional, std::variant and maybe even for std::disjunction.

File: include/unifex/config.hpp.in [hunk 1]
```diff
@@ -228,7 +228,7 @@
 #elif defined(__GNUC__) && __GNUC__ >= 6
 #define UNIFEX_IS_SAME(...) __is_same_as(__VA_ARGS__)
 #elif UNIFEX_CXX_TRAIT_VARIABLE_TEMPLATES
-#define UNIFEX_IS_SAME(...) std::is_same_v<__VA_ARGS__>
+#define UNIFEX_IS_SAME(...) std::std::is_same_v<__VA_ARGS__>
```
Comment (ispeters) [inline include/unifex/config.hpp.in:231]: Perhaps we should delete `UNIFEX_CXX_TRAIT_VARIABLE_TEMPLATES` instead of updating what happens when it's set.

Comment (MarkCMann) [reply include/unifex/config.hpp.in:231]: Is there a downside to removing these macros?

**Pull request merged**